ports/esp32/machine_timer.c

```diff
@@ -113,6 +115,12 @@ STATIC mp_obj_t machine_timer_make_new(const mp_obj_type_t *type, size_t n_args,
     self->next = MP_STATE_PORT(machine_timer_obj_head);
     MP_STATE_PORT(machine_timer_obj_head) = self;
 
+    if (n_args > 0 || n_kw > 0) {
+        mp_map_t kw_args;
+        mp_map_init_fixed_table(&kw_args, n_kw, args + n_args);
+        machine_timer_init_helper(self, n_args - 1, args + 1, &kw_args);
```

Comment:
will also need to call this init helper in the case above where the timer is already created (the bit that does `return t`)

Reply:
OK, Done. It required a little bit more of code line shuffling.